frame/examples/basic/src/tests.rs

```diff
@@ -21,19 +21,19 @@ use crate::*;
 use frame_support::{
 	assert_ok,
 	dispatch::{DispatchInfo, GetDispatchInfo},
+	macro_magic::use_attr,
 	traits::{ConstU64, OnInitialize},
 };
-use sp_core::H256;
+
 // The testing primitives are very useful for avoiding having to work with signatures
 // or public keys. `u64` is used as the `AccountId` and no `Signature`s are required.
-use sp_runtime::{
-	testing::Header,
-	traits::{BlakeTwo256, IdentityLookup},
-	BuildStorage,
-};
+use sp_runtime::{testing::Header, BuildStorage};
 // Reexport crate as its pallet name for construct_runtime.
 use crate as pallet_example_basic;
 
+#[use_attr]
```

Comment:
That this attribute is required, is not documented anywhere!

Can we not make `derive_impl` a "fake" macro that generates:

```
#[use_attr]
use frame_support::real_derive_impl;
#[real_derive_impl(frame_system::preludes::testing::TestDefaultConfig as frame_system::pallet::DefaultConfig)]
```

Something along these lines? Or do whatever `use_attr` is doing internally in the macro? So we don't need this

Reply:
But I didn't looked that much into the details if we can achieve this.

Reply:
This will work fine when `#[cfg(accessible(..))]` https://github.com/rust-lang/rust/issues/64797 is stabilized, but until then there isn't a good solution because having two such statements in the same file, it expands to a use so you'll get a double use.

```rust
#[use_attr]
use frame_support::real_derive_impl;
```
expands to:
```rust
use frame_support::real_derive_impl;
use frame_support::__export_tokens_tt_real_derive_impl; // this is decl macro
```

For this reason, glob imports like `frame_support::*` also work well because the export_tokens macro gets brought into scope for free, however people don't always want to glob import so I provide this syntax.

`#[use_attr]` is documented within `macro_magic` so the language server should have it. That said, I will add a note about it in the `derive_impl` docs 👍🏻

Reply:
TLDR: no, because then you can't use `derive_impl` twice in the same file